### ISSUE TITLE
Hyprland on Nvidia (vaapi): Failed to create vaapi frame context for capture surfaces of format

### ISSUE THREAD

**Issue by yuufox** [russelltg/wl-screenrec#79]
Hello!
My configuration:
```
Operating System:  Endeavour OS
Kernel: Linux 6.9.7-arch1-1
Architecture: x86-64
DE: Hyprland
GPU: NVIDIA RTX 3080
```
Apps:
```
aur/wl-screenrec 0.1.4-0 
extra/nvidia 555.58-2
extra/nvidia-utils 555.58-2
extra/libva-nvidia-driver 0.0.12-1
extra/hyprland 0.41.2-1
extra/ffmpeg 2:7.0.1-1
```

When running wl-screenrec --verbose
```
❯ wl-screenrec --verbose
08:25:42 [INFO] output probe complete: {ObjectId(wl_output@9, 9): Some(OutputInfo { name: "DP-1", loc: (0, 0), logical_size: (2560, 1440), size_pixels: (2560, 1440), refresh: Rational(164957/1000), fractional_scale: 1.0, output: WlOutput { id: ObjectId(wl_output@9, 9), version: 4, data: Some(ObjectData { .. }), backend: WeakBackend { inner: WeakInnerBackend { state: (Weak) } } }, transform: Normal })}
08:25:42 [INFO] Using output DP-1
Opening libva device from DRM device /dev/dri/renderD128
[AVHWFramesContext @ 0x582ba0ecc0c0] Failed to create surface: 14 (the requested RT Format is not supported).
[AVHWFramesContext @ 0x582ba0ecc0c0] Unable to allocate a surface from internal buffer pool.
failed to create encoder(s): Failed to create vaapi frame context for capture surfaces of format BGRZ 2560x1440
```
output of `wl-screenrec --no-hw` is basically the same.
```
❯ wl-screenrec --no-hw
Opening libva device from DRM device /dev/dri/renderD128
[AVHWFramesContext @ 0x5e988b689080] Failed to create surface: 14 (the requested RT Format is not supported).
[AVHWFramesContext @ 0x5e988b689080] Unable to allocate a surface from internal buffer pool.
failed to create encoder(s): Failed to create vaapi frame context for capture surfaces of format BGRZ 2560x1440

```

output of `vainfo`:
```
❯ vainfo
Trying display: wayland
vainfo: VA-API version: 1.21 (libva 2.22.0)
vainfo: Driver version: VA-API NVDEC driver [direct backend]
vainfo: Supported profile and entrypoints
      VAProfileMPEG2Simple            :	VAEntrypointVLD
      VAProfileMPEG2Main              :	VAEntrypointVLD
      VAProfileVC1Simple              :	VAEntrypointVLD
      VAProfileVC1Main                :	VAEntrypointVLD
      VAProfileVC1Advanced            :	VAEntrypointVLD
      VAProfileH264Main               :	VAEntrypointVLD
      VAProfileH264High               :	VAEntrypointVLD
      VAProfileH264ConstrainedBaseline:	VAEntrypointVLD
      VAProfileHEVCMain               :	VAEntrypointVLD
      VAProfileVP8Version0_3          :	VAEntrypointVLD
      VAProfileVP9Profile0            :	VAEntrypointVLD
      VAProfileAV1Profile0            :	VAEntrypointVLD
      VAProfileHEVCMain10             :	VAEntrypointVLD
      VAProfileHEVCMain12             :	VAEntrypointVLD
      VAProfileVP9Profile2            :	VAEntrypointVLD
      VAProfileHEVCMain444            :	VAEntrypointVLD
      VAProfileHEVCMain444_10         :	VAEntrypointVLD
      VAProfileHEVCMain444_12         :	VAEntrypointVLD
```

LIBVA_DRIVER_NAME is set via hyprland.conf:
```
env = LIBVA_DRIVER_NAME,nvidia
```

Is there any idea how to resolve this issue?
Thanks in advance!

**Comment by DmitryHudrich** [comment]
same issue

**Comment by BrayanLope5** [comment]
Also having the same issue using ibva-nvidia-driver and proprietary drivers for an rtx 2070 super.

**Comment by russelltg** [comment]
This isn't going to work as-is; there is no vaapi encode driver for Nvidia. 

However, vulkan encode is supposed to work well on nvidia. If someone's feeling adventurous it'd be great if someone built wl-screenrec out of the `vulkan` brnach, against ffmpeg head, which has vulkan encode support, and passed `--experimental-vulkan`. 

It doesn't really work on ANV  yet, but I think that's mostly because of the drivers being in such early state. Nvidia has supported these extensions for a few years now....

**Comment by BrayanLope5** [comment]
Taking a closer look at the [vaapi implementation for nvidia](https://github.com/elFarto/nvidia-vaapi-driver) that we are using, I noticed I had missed the important part `VA-API implementation that uses NVDEC`, `NVDEC`, `NV`, `DEC`, nvidia decoder!!

Anyways, I would test the vulkan implementation but I have very little knowledge about git and cargo (?). I also don't know how to check if I have what is needed to use vulkan on my devices.

**Comment by russelltg** [comment]
No problem. Hold on a bit until distros ship ffmpeg 7.1, which is the first release to have vulkan encode, and let's hope that the bugs I'm seeeing are intel specific :).

**Comment by russelltg** [comment]
Will be fixed by #95 , move discussion there
